providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/file_watcher.py

```diff
@@ -0,0 +1,71 @@
+import json
+import logging
+import os
+import re
+import threading
+import time
+import typing
+
+import yaml
+
+from openfeature.event import ProviderEventDetails
+from openfeature.provider.provider import AbstractProvider
+
+from .flags import Flag
+
+
+class FileWatcherFlagStore:
+    def __init__(self, file_path: str, provider: AbstractProvider):
+        self.file_path = file_path
+        self.provider = provider
+        self.last_modified = 0.0
+        self.flag_data: dict[str, Flag] = {}
+        self.load_data()
+        self.thread = threading.Thread(target=self.refresh_file, daemon=True)
+        self.thread.start()
+
+    def shutdown(self) -> None:
+        pass
+
+    def get_flag(self, key: str) -> typing.Optional[Flag]:
+        return self.flag_data.get(key)
+
+    def refresh_file(self) -> None:
+        while True:
+            time.sleep(1)
+            logging.debug("checking for new flag store contents from file")
+            last_modified = os.path.getmtime(self.file_path)
+            if last_modified > self.last_modified:
+                self.load_data(last_modified)
+
+    def load_data(self, modified_time: typing.Optional[float] = None) -> None:
+        try:
+            with open(self.file_path) as file:
+                if self.file_path.endswith(".yaml"):
+                    data = yaml.safe_load(file)
+                else:
+                    data = json.load(file)
+
+                self.flag_data = self.parse_flags(data)
+                logging.debug(f"{self.flag_data=}")
+                self.provider.emit_provider_configuration_changed(
+                    ProviderEventDetails(flags_changed=list(self.flag_data.keys()))
+                )
+            self.last_modified = modified_time or os.path.getmtime(self.file_path)
+        except Exception:
```

Comment:
do we have a more specific set of exceptions we can handle here?

Reply:
I will hunt them down and add what I can find, what is the expected behavior for unexpected exceptions though?

I presume we would prefer the client and background thread does not crash as a rule so that apps can continue as usual with fallback behavior without downtime until the root cause is fixed

Reply:
Added sensible exception set, but kept the catch-all case in addition. Found [similar approach in openfeature](https://github.com/open-feature/python-sdk/blob/main/openfeature/client.py#L357-L389) for reference so I find this consistent

Reply:
Great, thanks. I'll let @federicobond resolve the conversation if he agrees but the change makes sense to me.

Reply:
Sure! I worry that unexpected exceptions may mask programmer errors here but I realize that I am looking at the problem from the perspective of an application developer.

In a library, the best DX may be to just swallow these kinds of errors.